synapse/appservice/__init__.py

```diff
@@ -263,8 +263,9 @@ async def is_interested_in_room(
             room_id, store, on_invalidate=cache_context.invalidate
         )
 
+    @cached(num_args=1, cache_context=True)
```

Comment:
I'm not sure this will work exactly as we're hoping -- `EventBase` equality is checked by it being the exact same object (i.e. if you built two `EventBase` instances with identical properties they aren't equal).

This might be OK, but I'm not 100% sure.

Reply:
Ah, good catch. Ideally we'd cache on the event's ID instead of `EventBase`. I don't suppose there's an easy way to do that with `@cached`, though I can just create an `LruCache` much as we do for `_get_event_cache`:

https://github.com/matrix-org/synapse/blob/e2e1d90a5e4030616a3de242cde26c0cfff4a6b5/synapse/storage/databases/main/events_worker.py#L240-L243

Reply:
From the [conversation in #synapse-dev](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$mflf7_3Je3Cm_wzGNMP_u_ShDFxb2qqoVqGFMkjC8NE?via=matrix.org&via=element.io&via=vector.modular.im), I believe we concluded on adding an `event_id` argument to `is_interested_in_event`. Doing so was simple, though updating the tests was slightly tedious.

Done in 453fdacb7!